core/extractors/__init__.py

```
@@ -0,0 +1,91 @@
+
+    if file_path_lower.endswith('.tex'):
+        if LaTeXExtractor is not None:
+            return LaTeXExtractor(**kwargs)
+        raise ImportError("LaTeXExtractor not available")
+
+    if file_path_lower.endswith(('.py', '.js', '.ts', '.go', '.rs', '.java', '.c', '.cpp', '.h')):
+        if CodeExtractor is not None:
+            return CodeExtractor(**kwargs)
+        if TreeSitterExtractor is not None:
+            return TreeSitterExtractor(**kwargs)
+        raise ImportError("No code extractor available")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)